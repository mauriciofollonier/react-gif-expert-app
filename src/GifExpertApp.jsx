import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Game of Thrones']);

    const onAddCategory = ( newCategory ) => {
        // Validar que las categorias sean distintas
        if ( categories.includes( newCategory ) ) return;

        setCategories([ newCategory, ...categories ]);
        //setCategories( cat => [...categories, newCategory]);
    }

  return (
    <>
        <h1>Gift Expert App</h1>

        <AddCategory 
            // setCategories = { setCategories }
            // Pasa la func setCategories al componente
            // y asi puede modificar el estado categories
            // en el otro componente (AddCategory).
            onNewCategory = { (value) => onAddCategory(value)}
            // Esta prop recibe el newValue y lo usa para
            // cambiar el estado aca, con setCategories.
            // RECIBE UN EVENTO DEL COMPONENTE HIJO.
            />
        {
        categories.map( (category) => (
            <GifGrid 
            key= { category }
            category = { category }
        /> 
        ))
        }
    </>
  )
};
