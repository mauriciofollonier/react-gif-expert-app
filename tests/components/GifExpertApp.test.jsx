import { render, screen, fireEvent } from "@testing-library/react"
import { GifExpertApp } from "../../src/GifExpertApp"

describe('Pruebas en componente <GifExpertApp/>', () => { 

    test('Debe hacer match con el snapshot', () => {
      
        const { container } = render( <GifExpertApp/> );
        expect( container ).toMatchSnapshot();
    });
    test('Debe modificar el input y ser valido (mas de un caracter)', () => {
        
        render( <GifExpertApp/> );

        const input = screen.getByRole('textbox');
        fireEvent.input( input ,{ target: {value: 'House of the Dragon'} } );

        expect( input.value ).toBe('House of the Dragon')
        screen.debug()
    })
    
    
 })