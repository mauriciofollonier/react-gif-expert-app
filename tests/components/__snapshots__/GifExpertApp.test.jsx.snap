// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifExpertApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gift Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Game of Thrones
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
